File: backend-ts/src/entities/Goal.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
  UpdateDateColumn,
  ManyToOne,
  Index
} from 'typeorm';
import { User } from './User';

@Entity()
@Index(['userId'])
export class Goal {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column()
  userId: string;

  // Basic info
  @Column({ length: 255 })
  name: string;

  @Column({ length: 255 })
  nameHe: string;

  @Column({ type: 'text' })
  description: string;

  @Column({ type: 'text' })
  descriptionHe: string;

  @Column({ length: 100 })
  category: 'consistency' | 'performance' | 'strength' | 'endurance' | 'weight_loss';

  // Target details
  @Column({ length: 100 })
  targetType: 'sessions' | 'reps' | 'distance' | 'weight' | 'duration' | 'calories';

  @Column()
  targetValue: number;

  @Column({ length: 50 })
  timePeriod: 'daily' | 'weekly' | 'monthly' | 'yearly';

  // Gamification
  @Column({ default: 0 })
  pointsReward: number;

  @Column({ default: true })
  isActive: boolean;

  // Timestamps
  @CreateDateColumn()
  createdAt: Date;

  @UpdateDateColumn()
  updatedAt: Date;

  // Relationships
  @ManyToOne(() => User, user => user.goals)
  user: User;
}

File: _archive/python-backend-20251019-113536/app/schemas/goal.py
"""
Pydantic schemas for goal data validation
"""

from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime
import uuid

class GoalBase(BaseModel):
    """Base goal schema"""
    name: str = Field(..., description="Goal name in English")
    name_he: str = Field(..., description="Goal name in Hebrew")
    description: str = Field(..., description="Goal description in English")
    description_he: str = Field(..., description="Goal description in Hebrew")
    category: str = Field(..., description="Goal category (e.g., consistency, performance)")
    target_type: str = Field(..., description="Type of activity to track (e.g., session, reps, distance)")
    target_value: int = Field(..., description="Numerical target for the goal")
    time_period: str = Field(..., description="Time frame for the goal (e.g., weekly, monthly, daily)")
    points_reward: int = Field(..., description="Points awarded upon completion")
    is_active: bool = Field(True, description="Whether the goal is active")

class GoalCreate(GoalBase):
    """Schema for creating a goal"""
    pass

class GoalUpdate(BaseModel):
    """Schema for updating a goal"""
    name: Optional[str] = None
    name_he: Optional[str] = None
    description: Optional[str] = None
    description_he: Optional[str] = None
    category: Optional[str] = None
    target_type: Optional[str] = None
    target_value: Optional[int] = None
    time_period: Optional[str] = None
    points_reward: Optional[int] = None
    is_active: Optional[bool] = None

class GoalResponse(GoalBase):
    """Schema for goal response"""
    id: uuid.UUID
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
